# HG changeset patch
# User Rik <rik@octave.org>
# Date 1412476752 25200
#      Sat Oct 04 19:39:12 2014 -0700
# Node ID d627d9c8adfdddf69115717b3aaf2274fc35d27a
# Parent  b85d4930528ddf7e87efb2fba1913b9aec5b01e6
ls_command.m: Overhaul function.

* ls_command.m: Make error message more succinct.  Add BIST test.

diff --git a/scripts/miscellaneous/ls_command.m b/scripts/miscellaneous/ls_command.m
--- a/scripts/miscellaneous/ls_command.m
+++ b/scripts/miscellaneous/ls_command.m
@@ -42,17 +42,17 @@ function old_cmd = ls_command (cmd)
   if (nargin == 0 || nargin == 1)
 
     old_cmd = __ls_command__;
 
     if (nargin == 1)
       if (ischar (cmd))
         __ls_command__ = cmd;
       else
-        error ("ls_command: expecting argument to be a character string");
+        error ("ls_command: argument must be a character string");
       endif
     endif
 
   endif
 
 endfunction
 
 
@@ -60,8 +60,10 @@ endfunction
 %! cmd = ls_command ();
 %! assert (ischar (cmd));
 %! if (ispc () && ! isunix ())
 %!   assert (cmd(1:3), "dir"); 
 %! else
 %!   assert (cmd(1:2), "ls"); 
 %! endif
 
+%!error <argument must be a character string> ls_command (123)
+
